pkg/controllers/awsloadbalancercontroller/deployment.go

```diff
@@ -2,13 +2,207 @@ package awsloadbalancercontroller
 
 import (
 	"context"
+	"fmt"
+	"sort"
+	"strings"
 
+	"github.com/google/go-cmp/cmp"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
 	albo "github.com/openshift/aws-load-balancer-operator/api/v1alpha1"
 )
 
+const (
+	appLabelName    = "app.kubernetes.io/name"
+	appName         = "aws-load-balancer-operator"
+	appInstanceName = "app.kubernetes.io/instance"
+)
+
 func (r *AWSLoadBalancerControllerReconciler) ensureControllerDeployment(ctx context.Context, namespace string, image string, sa *corev1.ServiceAccount, controller *albo.AWSLoadBalancerController) (*appsv1.Deployment, error) {
-	return nil, nil
+	deploymentName := fmt.Sprintf("aws-load-balancer-operator-%s", controller.Name)
+	desired := desiredDeployment(deploymentName, namespace, image, r.VPCID, r.ClusterName, controller, sa)
+	err := controllerutil.SetOwnerReference(controller, desired, r.Scheme)
+	if err != nil {
+		return nil, fmt.Errorf("failed to set owner reference on deployment %s: %v", deploymentName, err)
+	}
+	exists, current, err := r.currentControllerDeployment(ctx, deploymentName, namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get existing deployment %s: %w", deploymentName, err)
+	}
+	if !exists {
+		err = r.createDeployment(ctx, desired)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create deployment %s: %w", deploymentName, err)
+		}
+		_, current, err = r.currentControllerDeployment(ctx, deploymentName, namespace)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get new deployment %s: %w", deploymentName, err)
+		}
+	}
+	return r.updateDeployment(ctx, current, desired)
+}
+
+func desiredDeployment(name, namespace, image string, vpcID, clusterName string, controller *albo.AWSLoadBalancerController, sa *corev1.ServiceAccount) *appsv1.Deployment {
+	d := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
```

Comment:
If the users want to modify the placement they can modify the deployment directly. Since that is the approach we are taking for resources I don't see why it shouldn't be the same for placement as well.